plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakePartitioning.java

```diff
@@ -140,11 +141,49 @@ public void testReadAllTypes()
     }
 
     @Test
-    public void testPartitionsSystemTableDoesNotExist()
+    public void testPartitionsSystemTableExists()
     {
-        assertQueryFails(
-                "SELECT * FROM \"partitions$partitions\"",
-                ".*'delta\\.tpch\\.\"partitions\\$partitions\"' does not exist");
+        assertQuerySucceeds("SELECT * FROM \"partitions$partitions\"");
+    }
+
+    @Test
+    public void testReadAllTypesPartitionsSystemTable()
+    {
+        assertThat(
+                query("SELECT " +
+                        "partition.p_string, " +
+                        "partition.p_byte, " +
+                        "partition.p_short, " +
+                        "partition.p_int, " +
+                        "partition.p_long, " +
+                        "partition.p_decimal, " +
+                        "partition.p_boolean, " +
+                        "partition.p_float, " +
+                        "partition.p_double, " +
+                        "partition.p_date, " +
+                        // H2QueryRunner does not support TIMESTAMP WITH TIME ZONE,
+                        // so instead we convert the TIMESTAMP WITH TIME ZONE from Delta to a string representation and compare that.
+                        "CAST(partition.p_timestamp AS VARCHAR), " +
```

Comment:
The current logic doesn't depend on H2. Please remove the comment and remove redundant cast.

Reply:
Cast is not redundant:

```
[ERROR] Failures: 
[ERROR]   TestDeltaLakePartitioning.testReadAllTypesPartitionsSystemTable:170 
Multiple Failures (1 failure)
-- failure 1 --
[Rows for query [SELECT partition.p_string, partition.p_byte, partition.p_short, partition.p_int, partition.p_long, partition.p_decimal, partition.p_boolean, partition.p_float, partition.p_double, partition.p_date, partition.p_timestamp, file_count, total_size FROM "partitions$partitions" LIMIT 1 ]] 
Expecting actual:
  (Alice, 123, 12345, 123456789, 1234567890123456789, 12345678901234567890.123456789012345678, true, 3.1415927, 3.141592653589793, 2014-01-01, 2014-01-01T23:00:01.123Z[UTC], 30, 136080)
to contain exactly in any order:
  [(Alice, 123, 12345, 123456789, 1234567890123456789, 12345678901234567890.123456789012345678, true, 3.1415927, 3.141592653589793, 2014-01-01, 2014-01-01 23:00:01.123 UTC, 30, 136080)]
elements not found:
  (Alice, 123, 12345, 123456789, 1234567890123456789, 12345678901234567890.123456789012345678, true, 3.1415927, 3.141592653589793, 2014-01-01, 2014-01-01 23:00:01.123 UTC, 30, 136080)
and elements not expected:
  (Alice, 123, 12345, 123456789, 1234567890123456789, 12345678901234567890.123456789012345678, true, 3.1415927, 3.141592653589793, 2014-01-01, 2014-01-01T23:00:01.123Z[UTC], 30, 136080)
at QueryAssertions$ResultAssert.lambda$matches$0(QueryAssertions$ResultAssert.java:746)
```

Reply:
Please update VALUES query.